src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpRecommendCaseInsensitiveStringComparisonFixer.cs

```diff
@@ -17,70 +17,92 @@
             mainInvocationInstance = null;
 
             if (invocationExpression.Expression is MemberAccessExpressionSyntax memberAccessExpression)
             {
-                ExpressionSyntax internalExpression = memberAccessExpression.Expression is ParenthesizedExpressionSyntax parenthesizedExpression ?
-                    parenthesizedExpression.Expression :
-                    memberAccessExpression.Expression;
+                ExpressionSyntax internalExpression = memberAccessExpression.Expression;
+                while (internalExpression is ParenthesizedExpressionSyntax parenthesizedExpression)
+                {
+                    internalExpression = parenthesizedExpression.Expression;
+                }
 
-                if (internalExpression is InvocationExpressionSyntax internalInvocationExpression &&
-                    internalInvocationExpression.Expression is MemberAccessExpressionSyntax internalMemberAccessExpression)
+                if (leftOffendingMethod != null &&
+                    internalExpression is InvocationExpressionSyntax internalInvocationExpression &&
+                    internalInvocationExpression.Expression is MemberAccessExpressionSyntax internalMemberAccessExpression &&
+                    internalMemberAccessExpression.Name.Identifier.Text == leftOffendingMethod)
                 {
+                    // We know we have an instance invocation that is an offending method, retrieve just the instance
                     mainInvocationInstance = internalMemberAccessExpression.Expression;
                 }
                 else
                 {
                     mainInvocationInstance = memberAccessExpression.Expression;
-                    isChangingCaseInArgument = true;
                 }
             }
 
-            foreach (ArgumentSyntax node in invocationExpression.ArgumentList.Arguments)
-            {
-                string? argumentName = node.NameColon?.Name.Identifier.ValueText;
-                isAnyArgumentNamed |= argumentName != null;
+            List<SyntaxNode> arguments = new();
+            bool isAnyArgumentNamed = false;
 
-                ExpressionSyntax argumentExpression = node.Expression is ParenthesizedExpressionSyntax argumentParenthesizedExpression ?
-                    argumentParenthesizedExpression.Expression :
-                    node.Expression;
+            foreach (IArgumentOperation arg in mainInvocationOperation.Arguments)
+            {
+                SyntaxNode newArgumentNode;
 
-                MemberAccessExpressionSyntax? argumentMemberAccessExpression = null;
-                if (argumentExpression is InvocationExpressionSyntax argumentInvocationExpression)
+                // When accessing the main invocation operation arguments, the bottom operation is retrieved
+                // so we need to go up until we find the actual argument syntax ancestor, and skip through any
+                // parenthesized syntax nodes
```

Comment:
nit: would be good if you can write example of such case.

Reply:
+1, actual examples here from the corresponding test would be helpful.

Reply:
They are already there. In `RecommendCaseInsensitiveStringComparison.Base.Tests.cs`, the test cases are named `DiagnosedAndFixedParenthesizedComplexCasesData`.